### PR TITLE
feat: Added server bootstrap and improvements.

### PR DESCRIPTION
**Pull request by yusinto**
This PR demoes the use of the node sdk handing over bootstrap values to the react sdk. However, because the LDProvider is client-only, ssr is not possible :(.

### CODE REVIEW

File: instrumentation.ts
Comment (yusinto) [inline instrumentation.ts:1]: callout: Use the register hook to run server init code.

File: launchDarkly/server/initNodeSdk.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+export const initNodeSdk = async () => {
+  if (process.env.NEXT_RUNTIME === 'nodejs') {
+    const sdk = await import('@launchdarkly/node-server-sdk');
+    const nodeClient = sdk.init(process.env.LAUNCHDARKLY_SDK_KEY ?? '');
+
+    // HACK: save this globally so we can access it from the app
+    global.nodeClient = nodeClient;
```
Comment (yusinto) [inline launchDarkly/server/initNodeSdk.ts]: callout: If you have a better suggestion other than global lmk. I tried and this is the only thing that works.

File: launchDarkly/client/LDProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+export function LDProvider({
+  children,
+  context,
+  options,
+}: {
+  children: ReactNode;
+  context: LDContext;
+  options: LDOptions;
+}) {
+  const Provider = use(
```
Comment (yusinto) [inline launchDarkly/client/LDProvider.tsx]: This is so cool!

File: components/ldprovider.tsx
Comment (yusinto) [inline components/ldprovider.tsx:1]: Moved under `launchDarkly/client/LDProvider`

File: app/layout.tsx [hunk 3]
```diff
@@ -1,50 +1,53 @@
...
+  // const flag = await nodeClient.variation('flag-key', context, false);
 
   return (
     <html lang="en">
       <body className={inter.className}>
         <Suspense fallback={<div>Loading...</div>}>
-          <AsyncLDProvider context={context}>{children}</AsyncLDProvider>
+          <LDProvider
+            context={context}
+            options={{ bootstrap: JSON.parse(JSON.stringify(bootstrap)) }}
```
Comment (yusinto) [inline app/layout.tsx:45]: This works! You can test by inspecting the network tab on the browser debugger and see there's no initial call for flags.

Comment (codyde) [reply app/layout.tsx:45]: This is so awesome and going to be such a low effort performance boost. Great thing is that the all flags call doesn't trigger evaluation events either, so theres no hit on experimentation.

We should double check with them though for client side experiments that get the values first from this method, is it still going to track events successfully? I think yes, just want to confirm. @mike-zorn or @cdelst?

File: launchDarkly/client/LDProvider.tsx (hunk 2 above)
Comment (codyde) [reply launchDarkly/client/LDProvider.tsx]: Yessss - slight tweak here on this, in the recent version of Next and NextJS 15 - use will throw an error in console (but still work fine). 

image

We want to do something like below instead. This moves the promise out and into the proper place. I'll push an update. 

```javascript 
'use client';

import { LDOptions } from 'launchdarkly-js-client-sdk';
import { LDContext } from 'launchdarkly-js-sdk-common';
import { asyncWithLDProvider } from 'launchdarkly-react-client-sdk';
import { ReactNode, use, useMemo } from 'react';

export function LDProvider({
  children,
  context,
  options,
}: {
  children: ReactNode;
  context: LDContext;
  options: LDOptions;
}) {
  const Provider = useMemo(
    () =>
      asyncWithLDProvider({
        clientSideID: process.env.NEXT_PUBLIC_LD_CLIENT_SDK_KEY ?? '',
        context,
        options,
      }),
    [context, options],
  );

  const LDClientProvider = use(Provider);

  return <LDClientProvider>{children}</LDClientProvider>;
}
```

File: launchDarkly/server/initNodeSdk.ts (hunk 1 above)
Comment (codyde) [reply launchDarkly/server/initNodeSdk.ts]: I think this is the best way - I was trying a few different things, but the ergonomic of yours is really good - and prevents them having to initiate a server client over and over again. 

How would you feel about calling it something different? Like LDserverClient or something like that? I just want to make the ergonomic easy for a customer to use - and nodeClient sounds much more like part of the core node process.

I'm very open to back and forth debate on this!

File: components/ldprovider.tsx
Comment (codyde) [reply components/ldprovider.tsx:1]: I like this, better organization. Very supportive.

File: launchDarkly/server/initNodeSdk.ts (hunk 1 above)
Comment (yusinto) [reply launchDarkly/server/initNodeSdk.ts]: Ok I'm renaming to `ldNodeClient` as discussed on slack.

File: app/layout.tsx (hunk 3 above)
Comment (yusinto) [reply app/layout.tsx:45]: On the client side, `useFlags` invokes `.variation` which tracks events correctly. I think we're good here.

**Pull request merged**